lib/testcase/sparql/TestCaseQueryEvaluation.ts

```diff
@@ -75,20 +76,23 @@ export class TestCaseQueryEvaluationHandler implements ITestCaseHandler<TestCase
    * @return {Promise<IQueryResult>} A promise resolving to a SPARQL query result.
    */
   public static async parseSparqlResults(type: 'json' | 'xml', data: NodeJS.ReadableStream): Promise<IQueryResult> {
-    let parser;
     let booleanPromise: Promise<boolean>;
     let bindingsStream: NodeJS.ReadableStream;
     if (type === 'json') {
-      parser = new SparqlJsonParser({ prefixVariableQuestionMark: true });
+      const parser = new SparqlJsonParser({ prefixVariableQuestionMark: true });
       booleanPromise = parser.parseJsonBooleanStream(data);
       bindingsStream = parser.parseJsonResultsStream(data);
     } else {
-      parser = new SparqlXmlParser({ prefixVariableQuestionMark: true });
+      const parser = new SparqlXmlParser({ prefixVariableQuestionMark: true });
       booleanPromise = parser.parseXmlBooleanStream(data);
       bindingsStream = parser.parseXmlResultsStream(data);
     }
     const bindingsPromise: Promise<[RDF.Variable[], { [variable: string]: RDF.Term }[]]> = Promise.all([
-      new Promise<RDF.Variable[]>((resolve) => bindingsStream.on('variables', resolve)),
+      new Promise<RDF.Variable[]>((resolve) => {
+        bindingsStream
+            .on('variables', resolve)
+            .on('end', _ => resolve([]))
```

Comment:
As discussed on gitter, I suspect this not to be necessary, and this to be a bug in the sparql parsers.